### PR TITLE
New delete layer

### PR DESCRIPTION
**Pull request by viddo**
delete-layer

This implements the new delete-layer modal, essentially the same as existing delete-map/dataset modal but w/ different text and side-effects.

Regarding the solution I'll add comments inline in code to explain some details.

### CODE REVIEW

File: lib/assets/javascripts/cartodb/table/actions_menu.js [hunk 1]
```diff
@@ -780,17 +791,8 @@
...
-        // including the map layer
         this.remove_dlg.ok = function() {
-          // Destroy layer model
           layer_view.model.destroy({
-            success: function() {
-              self._removeLayer(cid);
-              if(self.map.layers.getDataLayers().length === 1) {
-                self.map.layers.last().set('visible', true);
-              }
-            },
```
Comment (viddo) [inline lib/assets/javascripts/cartodb/table/actions_menu.js]: Removed the need for keeping a ref to the datalayer's cid here, instead all side-effects are now handled in the `_removeLayer` method so it's consistent for both old and new dialog.

**Comment by viddo** [comment]
@xavijam Since you seemed to implemented the original I think you're most qualified to review too, have a look. I'll deploy on ded02 staging so you can test it

**Comment by viddo** [comment]
cc @saleiva for the visuals as usual

**Comment by saleiva** [comment]
+1

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3448/))

File: lib/assets/javascripts/cartodb/table/actions_menu.js (hunk 1 above)
Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/actions_menu.js]: Yep, we talked about it yesterday: :+1:

File: lib/assets/javascripts/cartodb/new_common/dialogs/delete_layer/delete_layer_view.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    this._initViews();
+    this._initBinds();
+  },
+
+  render_content: function() {
+    return this._panes.getActivePane().render().el;
+  },
+
+  _initViews: function() {
+    this._panes = new cdb.ui.common.TabPane({
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/new_common/dialogs/delete_layer/delete_layer_view.js]: Ok, so you have created "states-element" and you only show the element you need. Ok!
Do you like it more than render each time state has changed?

**Comment by xavijam** [comment]
Looks good dude! :sake:

**Pull request merged**